src/cmap/connect.ts

```diff
@@ -381,10 +415,81 @@ function makeConnection(options: ConnectionOptions, _callback: CallbackWithType<
     options.cancellationToken.once('cancel', cancellationHandler);
   }
 
-  socket.once(connectEvent, connectHandler);
+  if (existingSocket) {
+    process.nextTick(connectHandler);
+  } else {
+    socket.once(connectEvent, connectHandler);
+  }
+}
+
+function makeSocks5Connection(
+  options: MakeConnectionOptions & { proxyOptions: undefined },
+  proxyOptions: ProxyOptions,
+  callback: Callback<Stream>
+) {
+  const proxyHost = proxyOptions.host.includes(':') ? `[${proxyOptions.host}]` : proxyOptions.host;
+  const proxyPort = proxyOptions.port ?? 1080;
+  const hostAddress = HostAddress.fromString(`${proxyHost}:${proxyPort}`);
```

Comment:
Might make sense to move this logic to a helper on HostAddress, either a bespoke one or maybe add flag to fromString to do this `.includes(':')` logic.

Reply:
@nbbeeken `HostAddress` is public API – is this okay? If so, does `HostAddress.fromHostPort(host, port)` sound good?

Reply:
HostAddress is public type API since its accessible on some events I think but its not exported for use at runtime so changing existing behavior is free game, so I think enhancing fromString would work. but fromHostPort sounds good too!

Reply:
Sure, done :) I do think it should be `HostAddress.fromHostPort(host, port)` since doing something like `HostAddress.fromString(`${host}:${port}`, true /* lastColonSeparatesPortFromHost */)` feels like it’s a bit more convoluted than it needs to be.

`HostAddress.fromSrvRecord()` was something that was added in the meantime as well. I’ve made it call `HostAddress.fromHostPort()`, because that felt a bit cleaner – I don’t think it makes much of a difference because we require SRV records to point to hostnames.